### PR TITLE
Bump prettyprinter upper bound, migrate to github actions

### PR DESCRIPTION
**Pull request by jhrcek**
My ultimate goal is to eventually switch PI to GHC 8.10.3 which is available in recently released [lts-17.0](https://www.stackage.org/lts-17.0)

This will require some version bumping in some of our dependencies.

This PR fixes build failures of elm-street with lts-17.0 and at the same time migrates its CI from travis (no longer works - see below - infinitely waiting for response for travis) to github actions. Unfortunately this won't show any CI results until the ci config is merged. You can check in my repo that it's working fine: https://github.com/jhrcek/elm-street/pull/1/checks

### CODE REVIEW

File: frontend/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,11646 @@
...
+      "dev": true
+    },
+    "neo-async": {
+      "version": "2.6.2",
+      "resolved": "https://registry.npmjs.org/neo-async/-/neo-async-2.6.2.tgz",
+      "integrity": "sha512-Yd3UES5mWCSqR+qNT93S3UoYUkqAZ9lLg8a7g9rimsWmYGK8cVToA4/sF3RrshdyV3sAGMXVUmpMYOw+dLpOuw==",
+      "dev": true
+    },
+    "nice-try": {
+
```
Comment (turboMaCk) [inline frontend/package-lock.json:11646]: I thought that point of using create-elm app is that you don't need to maintain stuff like this.

File: .github/workflows/haskell-ci.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    - name: Setup Node.js environment
+      uses: actions/setup-node@v1
+
+    - name: Build and test
+      working-directory: frontend
+      run: |
+          npm ci
+          npx elm-app build
+          npx elm-app test
```
Comment (turboMaCk) [inline .github/workflows/haskell-ci.yml:123]: I would ditch npm ci honestly. We don't depend on any node stuff other than create-elm-app so I would simply let it be what it is.

```suggestion
          npm install -g create-elm-app
          elm-app build
          elm-app test
```

Comment (jhrcek) [reply .github/workflows/haskell-ci.yml]: Good point, I ditched it all.

**Comment by turboMaCk** [comment]
@jhrcek this will require github repo admin to change settings which is currently configured for requiring travis-ci check which is removed as part of this PR.

**Comment by jhrcek** [comment]
> this will require github repo admin to change settings which is currently configured for requiring travis-ci check which is removed as part of this PR.

I think @arbus could help us with this. Also it would be nice if he could disable merge commits and allow squashing of PR commits in the settings of this repo so we can have cleaner history like in PI.

**Comment by arbus** [comment]
@jhrcek Done!

**Comment by turboMaCk** [comment]
@jhrcek can you try to `git commit --allow-empty "empty"` and push the commit so we can se if ci is running?

**Comment by jhrcek** [comment]
Tried it but it doesn't seem to have done anything, no CI is showing up in the requirements.

**Comment by arbus** [comment]
Do I need to do anything to enable github actions?

**Comment by jhrcek** [comment]
According to [this](https://docs.github.com/en/github/administering-a-repository/disabling-or-limiting-github-actions-for-a-repository) "By default, GitHub Actions is enabled on all repositories and organizations." so I supposed we just need to merge it to get the ci config available on master?

**Review by turboMaCk — APPROVED** [review]
LGTM. I guess we don't risk that much by just merging this.

**Comment by jhrcek** [comment]
@turboMaCk your review doesn't seem to count as you probably don't have write access.
We'll need @arbus to approve / merge this.

**Comment by turboMaCk** [comment]
> your review doesn't seem to count

**Comment by arbus** [comment]
Write access has been granted

**Comment by turboMaCk** [comment]
@arbus I think you should probably approve PRs in here yourself anyway since you have much better understanding of impact across projects.

Anyway from my side this looks good and we should not spent more time on this particular issue.

**Comment by Aman-Malhotra** [comment]
> > your review doesn't seem to count
> 
> 

😆 😆 😆 😆

**Comment by jhrcek** [comment]
@turboMaCk now that you have write access, but I don't can you merge, please? :laughing:

**Comment by arbus** [comment]
You both should...

**Comment by turboMaCk** [comment]
image

**Comment by jhrcek** [comment]
Maybe I have write access, but can't merge my own pr?
Screenshot from 2021-01-28 10-34-19

**Comment by arbus** [comment]
Try logging out and loggin in again?

**Comment by jhrcek** [comment]
Tried, no change.

**Comment by Aman-Malhotra** [comment]
Screenshot 2021-01-28 at 3 07 25 PM
just letting know that now I can merge too 😅

**Comment by jhrcek** [comment]
Now I can!

**Comment by arbus** [comment]
Good ol' turn it off and on again

**Pull request merged**